ipmininet/install/build_openr-rc-20190419-11514.sh

```diff
@@ -0,0 +1,301 @@
+#!/bin/bash
+
+set -exo pipefail
+### Install packages for Debian-based OS ###
+
+apt-get update && apt-get install -yq autoconf-archive bison build-essential cmake curl flex git gperf joe libboost-all-dev libcap-dev libdouble-conversion-dev libevent-dev libgflags-dev libgoogle-glog-dev libkrb5-dev libpcre3-dev libpthread-stubs0-dev libnuma-dev libsasl2-dev libsnappy-dev libsqlite3-dev libssl-dev libtool netcat-openbsd pkg-config sudo unzip wget python3-venv python-setuptools python3-setuptools python-pip python3-pip ccache
+apt-get install -yq gcc-'5' g++-'5'
+update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-'5' 40 --slave /usr/bin/g++ g++ /usr/bin/g++-'5'
+update-alternatives --config gcc
+
+PY_MAJOR=${PY_MAJOR:-2}
+
+if [ "$PY_MAJOR" -ne 2 ] && [ "$PY_MAJOR" -ne 3 ]; then
+    >&2 echo "PY_MAJOR must be set to either 2 or 3."
+    exit 1
+fi
+
+PY_BIN="python$PY_MAJOR"
+PIP_BIN="pip$PY_MAJOR"
+
+PY_VERSION="$($PY_BIN --version 2>&1)"
+PY_MINOR=$(sed -e 's/\(Python \)\([[:digit:]]\)\.\([[:digit:]]\)\(.*\)/\3/g' <<< "$PY_VERSION")
+PY_LIB_PATH="/usr/local/lib/python$PY_MAJOR.$PY_MINOR/site-packages"
+echo "PY_LIB_PATH: $PY_LIB_PATH"
+
+export CCACHE_DIR='/ccache' CC="ccache ${CC:-gcc}" CXX="ccache ${CXX:-g++}"
+### Diagnostics ###
+
+# Builder DebianSystemFBCodeBuilder(google/googletest:cmake_defines={u'BUILD_GTEST': u'ON', u'BUILD_SHARED_LIBS': u'OFF'}, gcc_version=u'5', facebook/zstd:git_hash=ShellQuoted(u'$(git describe --abbrev=0 --tags origin/master)'), hyperic/sigar:autoconf_options={u'CFLAGS': u'-fgnu89-inline'}, openr/build:cmake_defines={u'ADD_ROOT_TESTS': u'OFF'}, prefix=u'/usr/local', projects_dir=u'/usr/local/src', google/googletest:git_hash=u'release-1.8.1', wangle/wangle/build:cmake_defines={u'BUILD_TESTS': u'OFF'}, thom311/libnl:git_hash=u'libnl3_2_25', jedisct1/libsodium:git_hash=u'stable', ccache_dir=u'/ccache', make_parallelism=4, no1msd/mstch:git_hash=ShellQuoted(u'$(git describe --abbrev=0 --tags)'), zeromq/libzmq:git_hash=u'v4.2.5')
```

Comment:
What's the purpose of this line?
Replacing everything that is below if running on GCE? If so, you could use it as-is with some env detection?

Reply:
Are you asking about ccache or the comment?

- The former is a compiler cache that speeds up recompilation (I'm also not familiar with it).
- The latter is a comment with diagnostic purpose only, I guess. I belief the intention is to help you debugging the generated build script.

Reply:
About the comment. Re-reading, I confused myself thinking it was something else.